sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Upgrading.scala

```diff
@@ -334,9 +344,10 @@ object TypecheckUpgrades {
   private def checkDeleted[K, V](
       arg: Upgrading[Map[K, V]],
       handler: (K, V) => UpgradeError.Error,
+      ignore: (K, V) => Boolean = (_: K, _: V) => true,
   ): Try[(Map[K, Upgrading[V]], Map[K, V])] = {
     val (deletedV, existingV, newV) = extractDelExistNew(arg)
-    deletedV.headOption match {
+    deletedV.filter((kv: (K, V)) => ignore(kv._1, kv._2)).headOption match {
```

Comment:
```suggestion
    deletedV.view.filter((kv: (K, V)) => ignore(kv._1, kv._2)).headOption match {
```

to avoid computing the whole filtered map in some cases

Reply:
Added - very cool :D